### ISSUE TITLE
Freshly created dynamic reference variables cannot be read from immediately

### ISSUE THREAD

**Issue by mpmxyz** [Yellow-Dog-Man/Resonite-Issues#1394]
### Describe the bug?

I did some digging into the decompiled code - in short:
`DynamicVariableHelper.CreateReferenceVariable` does not call `IDynamicVariable.UpdateLinking` while `DynamicVariableHelper.CreateVariable` does it when it creates value variables.
This makes reading operations of reference variables fail unless a few update cycles have passed after their creation.

### To Reproduce

![example setup that creates dynvars of int/Slot and logs reading operations within the next 3 update cycles](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/8538547/7e94661b-b06a-4925-8629-faa52fe75a5d)
Basically follow the comments:
1. reset
3. execute test

### Expected behavior

Each 'log' line in the example should contain the written value.

### Resonite Version Number

Beta 2024.2.12.1430

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Desktop

### Log Files

[ANON - 2024.2.12.1430 - 2024-02-23 17_08_41.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/14387443/ANON.-.2024.2.12.1430.-.2024-02-23.17_08_41.log)

### Additional Context

Example object if you want to test yourself:
resrec:///U-TheAutopilot/R-3af28e74-88b7-4090-9510-d4a289faec23
